ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.cache
+
+import com.daml.caching.Cache
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.store.cache.StateCache.PendingUpdates
+import com.daml.scalautil.Statement.discard
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+
+/** This class is a wrapper around a Caffeine cache designed to handle
+  * correct resolution of concurrent updates for the same key.
+  */
+private[platform] case class StateCache[K, V](cache: Cache[K, V], metrics: Metrics)(implicit
+    ec: ExecutionContext
+) {
+  private val logger: ContextualizedLogger = ContextualizedLogger.get(getClass)
+  private[store] val pendingUpdates = mutable.Map.empty[K, PendingUpdates]
```

Comment:
Why is this scoped to `store` and not to `cache` where the tests reside?

Reply:
Thanks. Changed